backend/src/User/user.service.ts

```diff
@@ -37,6 +39,59 @@ class Service {
         };
     }
 
+    public async sendForgotPassword(email: string) {
+        const user = await UserModel.findByEmailAuth(email);
+        if (!user) {
+            return { status: false, error: 'Email does not match any account.' };
+        }
+
+        // Create token that expires in 1 hour
+        const token = crypto.randomBytes(20).toString('hex');
+        user.resetPasswordToken = token;
+        user.resetPasswordExpires = Date.now() + 3600000;
+
+        UserModel.updateById(user.userId, user);
+
+        const success = sendEmail.sendPasswordRecoveryEmail(user.email, token);
+        if (!success) {
+            return {
```

Comment:
We should log an error here

Reply:
Already logging an error in sendEmail.sendPasswordRecoveryEmail